rsmtool/preprocessor.py

```diff
@@ -1440,6 +1440,14 @@ def filter_data(self,
                            "for the following features: "
                            "{}".format(', '.join(missing_features)))
 
+        # if ``length_column`` exists and is non-numeric, convert it to numeric;
+        # values that cannot be coerced to numeric will be set to ``np.nan``
+        if length_column and not np.issubdtype(df_filtered['length'], np.number):
```

Comment:
Can't we just do `all(np.isreal(df_filtered['length']))`?

Reply:
Hmm. It looks like this actually won't work due to [this](https://github.com/pandas-dev/pandas/issues/10766) issue.

```
In [1]: import numpy as np

In [2]: import pandas as pd

In [3]: df = pd.DataFrame({'col': [1, 2, 3, 'A', '1.0', 5]})

In [4]: np.isreal(df.col)
Out[4]: array([ True,  True,  True,  True,  True,  True], dtype=bool)
```

Reply:
For the features I think we just do the conversion without raising the warning, perhaps we can do the same here? Would save us a condition :)

Reply:
That's OK with me if it's OK with @desilinguist. :)

Reply:
Oh, that's so stupid numpy!